packages/assets-controllers/src/MultichainBalancesController/MultichainBalancesController.ts

```diff
@@ -174,22 +176,77 @@ export class MultichainBalancesController extends BaseController<
       (balanceUpdate: AccountBalancesUpdatedEventPayload) =>
         this.#handleOnAccountBalancesUpdated(balanceUpdate),
     );
-    // TODO: Maybe add a MultichainAssetsController:accountAssetListUpdated event instead of using the entire state.
-    // Since MultichainAssetsController already listens for the AccountsController:accountAdded, we can rely in it for that event
-    // and not listen for it also here, in this controller, since it would be redundant
+
     this.messagingSystem.subscribe(
-      'MultichainAssetsController:stateChange',
-      async (assetsState: MultichainAssetsControllerState) => {
-        for (const accountId of Object.keys(assetsState.accountsAssets)) {
-          await this.#updateBalance(
+      'MultichainAssetsController:accountAssetListUpdated',
+      async ({ assets }) => {
+        const newAccountAssets = Object.entries(assets).map(
+          ([accountId, { added }]) => ({
             accountId,
-            assetsState.accountsAssets[accountId],
-          );
-        }
+            assets: [...added],
+          }),
+        );
+        await this.#updateBalancesForAccounts(newAccountAssets);
```

Comment:
We should probably add a `#handleOnAccountAssetListUpdated` method for this one, to be consistent with the other handlers